edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest/EdgeDeploymentControllerTest.cs

```diff
@@ -0,0 +1,590 @@
+                }
+            };
+            var creatOptions = new CreatePodParameters(null, exposedPorts, hostConfigs, null, null);
+            var controller = this.CreateDeploymentController(deviceSelector, moduleLifeCycleManager, "pvname", "storagename");
+            string imageName = "test-image:1";
+            KubernetesModule km1 = this.CreateKubernetesModule(moduleName, creatOptions, imageName);
+            var labels = new Dictionary<string, string>
+            {
+                ["net.azure-devices.edge.deviceid"] = "deviceid",
+                ["net.azure-devices.edge.hub"] = "hostname",
+                ["net.azure-devices.edge.module"] = moduleName
+            };
+
+            await this.client.AddModuleDeploymentAsync(moduleName, labels, null);
+            moduleLifeCycleManager.SetModules(moduleName);
+            await controller.DeployModulesAsync(ModuleSet.Empty, ModuleSet.Create(km1));
+
+            V1DeploymentList currentDeployments = await this.client.ListDeployments(deviceSelector);
+            V1ServiceAccountList currentServiceAccounts = await this.client.ListServiceAccounts(deviceSelector);
+            V1ServiceList currentServices = await this.client.ListServices(deviceSelector);
+            V1PersistentVolumeClaimList currentPvcList = await this.client.ListPeristentVolumeClaims();
+            Assert.Empty(currentDeployments.Items);
+            Assert.Empty(currentServiceAccounts.Items);
+            Assert.Empty(currentServices.Items);
+            Assert.Empty(currentPvcList.Items);
+        }
+
+        [Fact]
+        public async Task CheckIfDeploymentIsHealthyWithDeploymentDeletion()
+        {
+            var moduleName = "module-a";
+            var deviceSelector = $"{Kubernetes.Constants.K8sEdgeDeviceLabel}=deviceid,{Kubernetes.Constants.K8sEdgeHubNameLabel}=hostname";
+            var moduleLifeCycleManager = new DummyModuleIdentityLifecycleManager(
+                "hostname",
+                "gatewayhostname",
+                "deviceid",
+                moduleName,
+                new ConnectionStringCredentials("connectionString"));
+            var creatOptions = CreatePodParameters.Create();
+            var controller = this.CreateDeploymentController(deviceSelector, moduleLifeCycleManager, string.Empty, string.Empty);
+            string imageName = "test-image:1";
+            KubernetesModule km1 = this.CreateKubernetesModule(moduleName, creatOptions, imageName);
+            var labels = new Dictionary<string, string>
+            {
+                ["net.azure-devices.edge.deviceid"] = "deviceid",
+                ["net.azure-devices.edge.hub"] = "hostname",
+                ["net.azure-devices.edge.module"] = moduleName
+            };
+
+            await this.client.AddModuleDeploymentAsync(moduleName, labels, null);
+            moduleLifeCycleManager.SetModules(moduleName);
+            await controller.DeployModulesAsync(ModuleSet.Create(km1), ModuleSet.Create(km1));
```

Comment:
I'm not sure that testing DeployModulesAsync method with calling DeployModulesAsync in advance is a good approach. Isn't it better to prepare all required k8s resources with KubernetesClient in advance instead and use DeployModulesAsync only once in Act block?